src/nupic/engine/Region.cpp

```diff
@@ -61,13 +58,15 @@ Region::Region(std::string name, const std::string &nodeType,
   RegionImplFactory &factory = RegionImplFactory::getInstance();
   spec_ = factory.getSpec(nodeType);
 
+  // Dimensions start off as don't care but can be changed by user. dek 02/2019
+  // singleNodeOnly is ignored.
+  //dims_.push_back(1);
```

Comment:
is this comment valid?

Reply:
Yes, I am trying to change the behavior a little.  Let me try to explain this...

The original behavior is that if the following line is in the Spec (and all region implementations do except TestNode)
```
ns->singleNodeOnly = true;
```
then the dimension is set to `dim = {1};` which means the output buffer is single dimension and the Link must ask the region implementation how wide it is (by calling getNodeOutputElementCount( )). So basically the region's configuration parameters determine the width of the output buffer and the user cannot call region->setDimension(dim) to set a dimension.

What I wanted to do was allow a user to set dimensions on a region anyway.  That value would be placed in the SDR when the Link creates the output SDR.  The dimensions on an input SDR would be determined by the output to which it is connected.

This is not perfect, the region->setDimension(dim) will affect all outputs of a region and I think the user should be able specify which output on a region they want the dimension to be set on.

I have been struggling with this.  I have been thinking that perhaps we should have a discussion about what a dimension means and do we need it at all and if so, do we need it to be set outside of an algorithm.  See Issue #273.